boogie-routes/src/test/java/org/mitre/tdp/boogie/alg/facade/FluentRouteExpanderTest.java

```diff
@@ -107,6 +85,32 @@ void testMultipleStarTrans() {
     );
   }
 
+  /**
+   * Now we added a costr3 mock that has an ATL airport identifier
+   */
+  @Test
+  void testErrantStar() {
+    String route = "LBV.COSTR3.KMCO";
+
+    Fix lbv = fix("LBV", 26.828186111111112, -81.3914388888889);
+
+    COSTR3_DUP atlConstr3 = spy(COSTR3_DUP.INSTANCE);
+    doReturn("KATL").when(atlConstr3).airportIdentifier();
+    doReturn(ProcedureType.SID).when(atlConstr3).procedureType();
+
+    FluentRouteExpander expander = newExpander(
+        singletonList(lbv),
+        emptyList(),
+        singletonList(Airports.KMCO()),
+        List.of(COSTR3.INSTANCE, atlConstr3));
```

Comment:
<img width="981" alt="Screenshot 2025-04-17 at 2 19 20 PM" src="https://github.com/user-attachments/assets/4c9fb997-de81-477e-bfa6-429a2f980e8c" />

The `resolvedTokens` includes both my mocked ATL SID and the MCO STAR, but the `filteredResolvedTokens` correctly includes only the MCO STAR.